java-util/src/main/java/io/druid/java/util/common/granularity/PeriodGranularity.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.java.util.common.granularity;
+
+import com.google.common.base.Preconditions;
+import org.joda.time.Chronology;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Period;
+import org.joda.time.chrono.ISOChronology;
+
+public class PeriodGranularity {
+    private final Period period;
+    private final Chronology chronology;
+    private final long origin;
+    private final boolean hasOrigin;
+
+    private final boolean isCompound;
+
+    public PeriodGranularity(Period period, DateTime origin, DateTimeZone tz)
```

Comment:
is this class only for segments or queries?

If so, can we call this SegmentGranularity or QueryGranularity?

I prefer for there to be SegmentGranularity and QueryGranularity, and for us to get rid of the general class Granularity